src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,624 @@
+
+Chao1 and ACE are nonparametric estimators that describe diversity as the max number of expected species in the sample (species richness). They consider the proportion of species that have been observed before (“recaptured”) to those that are observed only once. Both Chao1 and ACE underestimate true richness at low sample sizes.
+
+On the other hand, Simpson and Shannon-Weaver, consider relative abundances and depends on both, species richness and the evenness (or equitableness), with which individuals are distributed among the different species. However, both metrics have specific biases. The Shannon-Weaver index places a greater weight on species richness, whereas the Simpson index considers evenness more than richness in its measurement.
+
+More discussion and examples illustrating how sample size and normalization methods can affect alpha diversity metrics in references 4 and 5.
+
+In this example, we're going to calculate the diversity measures for our data with no normalization, our TSS normalized data, and our rarefied data.
+
+1. Now, let's use the `addAlpha` method to compute the alpha diversity indices on our non-normalized data, our rarefied data, and our tss normalized data. We also add a `SampleID` colData column to `filtered_tse` to help with out plotting too. For this example we'll use the `shannon_diversity` index to show alpha diversity, though [see here](https://microbiome.github.io/mia/reference/addAlpha.html) for more index options.
+
+```{r}
+# Add SampleID to colData
+filtered_tse$SampleID <- colnames(filtered_tse)
+
+# Compute alpha diversity on normal counts data
+filtered_tse <- addAlpha(filtered_tse, assay.type = "counts", index = "shannon_diversity", 
+                         name = "shannon_diversity_normal")
+
+# Compute alpha diversity on rarefied data
+filtered_tse <- addAlpha(filtered_tse, assay.type = "rarefy", index = "shannon_diversity", 
+                         name = "shannon_diversity_rarefy")
+```
+
+2. Plot alpha diversity with `plotColData`.
+
+```{r}
+# Plot with no normalization
+alpha_diversity_no_norm <- plotColData(filtered_tse, x = "SampleID", y = "shannon_diversity_normal", 
+                                       colour_by = "SampleID") +
+                           labs(y = "Shannon Diversity Index") +
+                           theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 6))
+                           
+# Plot with rarefied data
+alpha_diversity_rarefy <- plotColData(filtered_tse, x = "SampleID", y = "shannon_diversity_rarefy", 
+                                       colour_by = "SampleID") +
+                           labs(y = "Shannon Diversity Index") +
+                           theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 6))
```

Comment:
Is it possible to do this with lapply, only switching the y? 

Like

lapply(c("shannon_diversity_normal", "shannon_diversity_rarefy"), function (y) {plotColData(... y=y)})